### PR TITLE
[sdk-metrics] Promote cardinality limit view API from experimental to stable

### LINKED ISSUE #5444: Promote cardinality limit view API from experimental to stable
We have an experimental api now for setting cardinality limit via the view api (see: #5312 & #5328).

This issue is for tracking making this a stable feature.

### PR DESCRIPTION
**Pull request by xiang17**
Fixes #5444.

## Changes

Promote cardinality limit view API from experimental to stable. Updates README and obsolete flags for `MeterProviderBuilder.SetMaxMetricPointsPerMetricStream` to note that they are in stable.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable): The APIs were in `src\OpenTelemetry\.publicApi\Experimental\PublicAPI.Unshipped.txt`.

### CODE REVIEW

**Comment by xiang17** [comment]
What should I do with the PublicAPI files in Experimental folder?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/9f41eadf03f3dcc5e76c686b61fb39849f046312/src/OpenTelemetry/.publicApi/Experimental/PublicAPI.Unshipped.txt#L20-L21

File: src/OpenTelemetry/Metrics/Builder/MeterProviderBuilderExtensions.cs [hunk 1]
```diff
@@ -238,9 +238,7 @@ public static MeterProviderBuilder SetMaxMetricStreams(this MeterProviderBuilder
     /// <param name="meterProviderBuilder"><see cref="MeterProviderBuilder"/>.</param>
     /// <param name="maxMetricPointsPerMetricStream">Maximum number of metric points allowed per metric stream.</param>
     /// <returns>The supplied <see cref="MeterProviderBuilder"/> for chaining.</returns>
-#if EXPOSE_EXPERIMENTAL_FEATURES
-    [Obsolete("Use MetricStreamConfiguration.CardinalityLimit via the AddView API instead. This method will be removed in a future version.")]
-#endif
+    [Obsolete("Use MetricStreamConfiguration.CardinalityLimit via the AddView API instead. This method is removed in version 1.10.0.")]
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry/Metrics/Builder/MeterProviderBuilderExtensions.cs:241]:
```suggestion
    [Obsolete("Use MetricStreamConfiguration.CardinalityLimit via the AddView API instead. This method is marked as obsolete in version 1.10.0 and will be removed in a future version.")]
```

File: docs/diagnostics/experimental-apis/README.md [hunk 2]
```diff
@@ -58,3 +52,11 @@ Description: Metrics Exemplar Support
 Details: [OTEL1002](https://github.com/open-telemetry/opentelemetry-dotnet/blob/b8ea807bae1a5d9b0f3d6d23b1e1e10f5e096a25/docs/diagnostics/experimental-apis/OTEL1002.md)
 
 Released stable: `1.9.0`
+
+### OTEL1003
```
Comment (reyang) [inline docs/diagnostics/experimental-apis/README.md:56]: Does this need to be moved?

File: src/OpenTelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,11 @@ Notes](../../RELEASENOTES.md).
 
 ## Unreleased
 
+* The experimental APIs previously covered by `OTEL1003`
+  (`MetricStreamConfiguration.CardinalityLimit`) will now be part of the public
```
Comment (reyang) [inline src/OpenTelemetry/CHANGELOG.md:10]: will be -> is?

Comment (reyang) [reply src/OpenTelemetry/CHANGELOG.md:10]: I wonder how folks think about https://www.theregister.com/2024/10/08/linus_torvalds_grammar_complaint/?

File: docs/diagnostics/experimental-apis/README.md (hunk 2 above)
Comment (xiang17) [reply docs/diagnostics/experimental-apis/README.md:56]: I was simply following the convention of the OTEL1002 above which was released and moved from "Active" to "Inactive". Personally I think it'd also be helpful to have a record of what happened to previous experimental features.

File: src/OpenTelemetry/CHANGELOG.md (hunk 3 above)
Comment (xiang17) [reply src/OpenTelemetry/CHANGELOG.md:10]: This wording is also used in other experimental APIs in this CHANGELOG.

**Review by reyang — APPROVED** [review]
Left couple suggestions/questions that are non-blocking.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry/CHANGELOG.md:10]: I think "is" is better. Would be good to follow up and make changes everywhere. Not a blocker for this.

Comment (xiang17) [reply src/OpenTelemetry/CHANGELOG.md:10]: Created a PR #5940 for this.

**Pull request merged**